packages/@glimmer/runtime/lib/environment.ts

```diff
@@ -23,92 +25,86 @@ import DebugRenderTree from './debug-render-tree';
 export const TRANSACTION: TransactionSymbol = symbol('TRANSACTION');
 
 class TransactionImpl implements Transaction {
-  public scheduledInstallManagers: InternalModifierManager[] = [];
-  public scheduledInstallModifiers: unknown[] = [];
-  public scheduledUpdateModifierManagers: InternalModifierManager[] = [];
-  public scheduledUpdateModifiers: unknown[] = [];
-  public createdComponents: unknown[] = [];
-  public createdManagers: WithCreateInstance<unknown>[] = [];
-  public updatedComponents: unknown[] = [];
-  public updatedManagers: WithCreateInstance<unknown>[] = [];
-
-  didCreate(component: unknown, manager: WithCreateInstance) {
+  public scheduledInstallModifiers: ModifierInstance[] = [];
+  public scheduledUpdateModifiers: ModifierInstance[] = [];
+  public createdComponents: ComponentInstanceWithCreate[] = [];
+  public updatedComponents: ComponentInstanceWithCreate[] = [];
+
+  didCreate(component: ComponentInstanceWithCreate) {
     this.createdComponents.push(component);
-    this.createdManagers.push(manager);
   }
 
-  didUpdate(component: unknown, manager: WithCreateInstance) {
+  didUpdate(component: ComponentInstanceWithCreate) {
     this.updatedComponents.push(component);
-    this.updatedManagers.push(manager);
   }
 
-  scheduleInstallModifier(modifier: unknown, manager: InternalModifierManager) {
+  scheduleInstallModifier(modifier: ModifierInstance) {
     this.scheduledInstallModifiers.push(modifier);
-    this.scheduledInstallManagers.push(manager);
   }
 
-  scheduleUpdateModifier(modifier: unknown, manager: InternalModifierManager) {
+  scheduleUpdateModifier(modifier: ModifierInstance) {
     this.scheduledUpdateModifiers.push(modifier);
-    this.scheduledUpdateModifierManagers.push(manager);
   }
 
   commit() {
-    let { createdComponents, createdManagers } = this;
+    let { createdComponents, updatedComponents } = this;
 
     for (let i = 0; i < createdComponents.length; i++) {
-      let component = createdComponents[i];
-      let manager = createdManagers[i];
-      manager.didCreate(component);
+      let { manager, state } = createdComponents[i];
+      manager.didCreate(state);
     }
 
-    let { updatedComponents, updatedManagers } = this;
-
     for (let i = 0; i < updatedComponents.length; i++) {
-      let component = updatedComponents[i];
-      let manager = updatedManagers[i];
-      manager.didUpdate(component);
+      let { manager, state } = updatedComponents[i];
+      manager.didUpdate(state);
     }
 
-    let { scheduledInstallManagers, scheduledInstallModifiers } = this;
+    let { scheduledInstallModifiers, scheduledUpdateModifiers } = this;
 
-    let manager: InternalModifierManager, modifier: unknown;
+    // Prevent a transpilation issue we guard against in Ember
```

Comment:
Can you tweak this to mention the babel transform that we use and its option? That way we'll know if we can change it again in the future.

https://babeljs.io/docs/en/babel-plugin-transform-block-scoping.html#throwifclosurerequired